lib/email/receiver.rb

```diff
@@ -195,24 +196,33 @@ def log_and_validate_user(user)
     end
 
     def is_bounce?
-      return false unless @mail.bounced? || verp
+      @mail.bounced? || verp
+    end
 
+    def handle_bounce
       @incoming_email.update_columns(is_bounce: true)
 
       if verp && (bounce_key = verp[/\+verp-(\h{32})@/, 1]) && (email_log = EmailLog.find_by(bounce_key: bounce_key))
         email_log.update_columns(bounced: true)
-        email = email_log.user.try(:email).presence
+        user = email_log.user
+        email = user.try(:email).presence
```

Comment:
I know you just copied this line, but it's worth switching to the Safe Navigation Operator in the process.

`email = user&.email.presence`